cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/replication/BasicPullStrategyMockTest.java

```diff
@@ -170,29 +171,26 @@ public void testReplicationDocWithEmptyId() throws Exception {
             @Override
             public Object answer(InvocationOnMock invocation) throws Throwable {
                 JSONHelper jsonHelper = new JSONHelper();
-                FileReader fr = new FileReader(TestUtils.loadFixture("fixture/testReplicationDocWithEmptyId_changes.json"));
+                FileReader fr = new FileReader(TestUtils.loadFixture
+                        ("fixture/testReplicationDocWithEmptyId_changes.json"));
                 return jsonHelper.fromJson(fr, ChangesResult.class);
             }
         });
         when(mockRemoteDb.exists()).thenReturn(true);
-        Collection<String> revs = new ArrayList<String>();
-        revs.add("1-bd42b942b8b672f0289cf3cd1f67044c");
-        when(mockRemoteDb.getRevisions("", revs, new HashSet<String>(), false)).then(new Answer<Object>() {
+        when(mockRemoteDb.bulkGetRevisions((List<BulkGetRequest>)anyObject(), eq(false))).then(new Answer<Object>() {
             @Override
             public Object answer(InvocationOnMock invocation) throws Throwable {
-
-                return loadOpenRevsResponseFromFixture("testReplicationDocWithEmptyId_open_revs_1.json");
-
-            }
-        });
-        revs = new ArrayList<String>();
-        revs.add("1-13d33701a0954729ad029adf8fdc5a04");
-        when(mockRemoteDb.getRevisions("4d3b3f01362649d79b31d9092799a7e0", revs, new HashSet<String>(),false)).then(new Answer<Object>() {
-            @Override
-            public Object answer(InvocationOnMock invocation) throws Throwable {
-
-                return loadOpenRevsResponseFromFixture("fixture/testReplicationDocWithEmptyId_open_revs_2.json");
-
+                List<BulkGetRequest> requests = ((List<BulkGetRequest>)invocation.getArguments()[0]);
+                System.out.println(requests.get(0).id);
+                if (requests.get(0).id.equals("")) {
+                    // well this will never get called with "" because empty ids get skipped - should we fail?
```

Comment:
If our expected behaviour is that empty IDs should be skipped then yes, we should fail.